src/Microsoft.VisualStudio.Threading.Analyzers.CSharp/VSTHRD110ObserveResultOfAsyncCallsAnalyzer.cs

```diff
@@ -42,6 +40,112 @@ public override void Initialize(AnalysisContext context)
             context.RegisterSyntaxNodeAction(Utils.DebuggableWrapper(this.AnalyzeInvocation), SyntaxKind.InvocationExpression);
         }
 
+        private static bool IsAwaitable(ITypeSymbol returnedSymbol) =>
+            returnedSymbol.Name switch
+            {
+                Types.Task.TypeName
+                    when returnedSymbol.BelongsToNamespace(Types.Task.Namespace) => true,
+
+                Types.ConfiguredTaskAwaitable.TypeName
+                    when returnedSymbol.BelongsToNamespace(Types.ConfiguredTaskAwaitable.Namespace) => true,
+
+                Types.ValueTask.TypeName
+                    when returnedSymbol.BelongsToNamespace(Types.ValueTask.Namespace) => true,
+
+                Types.ConfiguredValueTaskAwaitable.TypeName
+                    when returnedSymbol.BelongsToNamespace(Types.ConfiguredValueTaskAwaitable.Namespace) => true,
+
+                _ => false,
+            };
+
+        private static bool IsCustomAwaitable(ITypeSymbol returnedSymbol)
+        {
+            // type has method: public T GetAwaiter()
+            IMethodSymbol? getAwaiterMethod = null;
+            ImmutableArray<ISymbol> members = returnedSymbol.GetMembers("GetAwaiter");
+            for (var i = 0; i < members.Length; ++i)
+            {
+                ISymbol member = members[i];
+                if (member.DeclaredAccessibility == Accessibility.Public
```

Comment:
I don't think public accessibility is a requirement, is it? It could be internal if the type were defined in this same assembly, I suspect.

We already have `IsAwaitableType` defined in the project, BTW. Can we just use that instead of these two new methods you've defined here?

Reply:
@AArnott you are right, I can see such method - will need to figure out how to use it (also I can see a lot of LINQ being used there, but let's keep that for another PR :sweat_smile:)

Also good catch with `public` 🙇 !